lib/index.js

```diff
@@ -1,56 +1,67 @@
-var handler = require('./handler');
-var AWS = require('aws-sdk');
-var datefmt = require('datefmt');
-var Hoek = require('hoek');
-var slug = require('slug');
+const Boom = require('boom');
+const Joi = require('joi');
+const slug = require('slug');
+const s3put = require('s3put');
 
 // Forces slug to use url encoding
 slug.defaults.mode = 'rfc3986';
 
-module.exports = function(plugin, options, next) {
-
+module.exports = (plugin, options, next) => {
```

Comment:
can you run a joi check to make sure either key/secret or profile is set? as well as bucket?